### PR TITLE
transition to markdown, readme, and derivatives

### PR DESCRIPTION
**Pull request by dericed**

### CODE REVIEW

File: ffv1.md [hunk 1]
```diff
@@ -0,0 +1,960 @@
...
+AVI RIFF File Format <https://msdn.microsoft.com/en-us/library/windows/desktop/dd318189%28v=vs.85%29.aspx>
+
+Information technology Coding of audio-visual objects Part 12: ISO base media file format <http://www.iso.org/iso/iso_catalogue/catalogue_tc/catalogue_detail.htm?csnumber=61988>
+
+NUT Open Container Format <http://www.ffmpeg.org/~michael/nut.txt>
+
+# Copyright
+
+
+Copyright 2003-2013 Michael Niedermayer \<michaelni@gmx.at\>\
```
Comment (TimothyGu) [inline ffv1.md]: Extra empty lines, and extra backslashes.

File: ffv1.md [hunk 2]
```diff
@@ -0,0 +1,960 @@
+<h1>FFV1 Video Codec Specification</h1>
+
+<h2>by Michael Niedermayer <michaelni@gmx.at></h2>
```
Comment (TimothyGu) [inline ffv1.md]:
```md
# FFV1 Video Codec Specification
## by Michael Niedermayer <michaelni@gmx.at>
```

File: ffv1.md [hunk 3]
```diff
@@ -0,0 +1,960 @@
...
+|<code>a + b, a - b</code>                 |
+|<code>a << b, a >> b</code>               |
+|<code>a < b, a <= b, a > b, a >= b</code> |
+|<code>a == b, a != b</code>               |
+|<code>a & b</code>                        |
+|<code>a &#124; b</code>                   |
+|<code>a && b</code>                       |
+|<code>a &#124;&#124; b</code>             |
+|<code>a ? b : c</code>                    |
+|<code>a = b, a += b, a -= b</code>        |
```
Comment (TimothyGu) [inline ffv1.md]: `<code></code>` → <code>``</code>

File: ffv1.md [hunk 4]
```diff
@@ -0,0 +1,960 @@
...
+
+## Range
+
+__a...b__ means any value starting from a to b, inclusive.
+
+## Bitstream functions
+
+**remaing_bits_in_bitstream( )** means the count of remaining bits after the current position in the bitstream. It is computed from the NumBytes value multiplied by 8 minus the count of bits already read by the bitstream parser.
```
Comment (TimothyGu) [inline ffv1.md]: s/remaing_bits_in_bitstream/remainng_bits_in_bitstream/

File: ffv1.md [hunk 5]
```diff
@@ -0,0 +1,960 @@
...
+
+For the purpose of the predictior and context, samples above the coded slice are assumed to be 0; samples to the right of the coded slice are identical to the closest left sample; samples to the left of the coded slice are identical to the top right sample (if there is one), otherwise 0.
+
+| | | | | | | | |
+|---|---|---|---|---|---|---|---|
+|0|0| |0|0|0| |0|
+|0|0| |0|0|0| |0|
+|0|0| |a|b|c| |c|
+|0|a| |d| |e| |e|
+|0|d| |f|g|h| |h|
```
Comment (TimothyGu) [inline ffv1.md]: Some pretty printing please. Also what's up with the empty table header row? Is there a way to make it disappear?

File: ffv1.md [hunk 6]
```diff
@@ -0,0 +1,960 @@
...
+$L_{0}=2^{8}B_{0}+B_{1}$
+
+
+$j_{0}=2$
+
+#### Range non binary values
+
+To encode scalar integers it would be possible to encode each bit separately and use the past bits as context. However that would mean 255 contexts per 8-bit symbol which is not only a waste of memory but also requires more past data to reach a reasonably good estimate of the probabilities. Alternatively assuming a Laplacian distribution and only dealing with its variance and mean (as in Huffman coding) would also be possible, however, for maximum flexibility and simplicity, the chosen method uses a single symbol to encode if a number is 0 and if not encodes the number using its exponent, mantissa and sign. The exact contexts used are best described by the following code, followed by some comments.
+
+```
```
Comment (TimothyGu) [inline ffv1.md]: Use

```
```c
```

File: ffv1.md [hunk 7]
```diff
@@ -0,0 +1,960 @@
...
+
+
+|k | bits | value|
+|:---:|:------------------------:|:-------:|
+|0 | 1 | 0|
+|0 | 001 | 2|
+|2 | 1 00 | 0|
+|2 | 1 10 | 2|
+|2 | 01 01 | 5|
+|any | 000000000000 10000000 | 139|
```
Comment (TimothyGu) [inline ffv1.md]: Vertical alignment. Also I'd use <code>``</code> for the bits to make them line up.

File: ffv1.md [hunk 8]
```diff
@@ -0,0 +1,960 @@
...
+In the case of a bitstream with version >= 2, a configuration record is stored in the the underlying container, at the track header level.
+It contains the parameters used for all frames.
+The size of the configuration record, NumBytes, is supplied by the underlying container.
+
+|                                                          |      |
+|----------------------------------------------------------|-----:|
+| ConfigurationRecord( NumBytes ) {                        |      |
+|    ConfigurationRecordIsPresent = 1                      |      |
+|    Parameters( )                                         |      |
+|    while( remaining\_bits\_in\_bitstream( ) \> 32)       |      |
```
Comment (TimothyGu) [inline ffv1.md]: Here and in a lot of places, please be consistent with white space. In this case, put a space after `32`.

File: ffv1.md [hunk 9]
```diff
@@ -0,0 +1,960 @@
...
+
+**NumBytes** is defined as the size, in bytes, of the codec_specific_data element as indicated in the “length” field of codec_specific_data
+
+## Frame
+
+|                                                        |
+|---------------------------------------------------|---:|
+|Frame( ) {                                         |type|
+|    keyframe                                       |  br|
+|    if( keyframe && !ConfigurationRecordIsPresent) |    |
```
Comment (TimothyGu) [inline ffv1.md]: `…Present )`

File: ffv1.md [hunk 10]
```diff
@@ -0,0 +1,960 @@
...
+
+|                                                                    |
+|------------------------------------------------------------|------:|
+|Slice( i ) {                                                | type  |
+|    if( version \> 2 )                                      |       |
+|        SliceHeader( i )                                    |       |
+|    if( colorspace\_type == 0) {                            |       |
+|        for( p = 0; p \< primary\_color\_count; p++ ) {     |       |
+|            Plane( p )                                      |       |
+|    } else if( colorspace\_type == 1) {                     |       |
```
Comment (TimothyGu) [inline ffv1.md]: ditto

File: ffv1.md [hunk 11]
```diff
@@ -0,0 +1,960 @@
...
+|       for( i = 1; i \< 256; i++ )                          |     |
+|           state\_transition\_delta[ i ]                    | sr  |
+|   colorspace\_type                                         | ur  |
+|   if( version \> 0 )                                       |     |
+|       bits\_per\_raw\_sample                               | ur  |
+|   chroma\_planes                                           | br  |
+|   log2( h\_chroma\_subsample )                             | ur  |
+|   log2( v\_chroma\_subsample )                             | ur  |
+|   alpha\_plane                                             | br  |
+|   if ( version \> 1 ) {                                    |     |
```
Comment (TimothyGu) [inline ffv1.md]: `if( version`. Ditto below.

File: ffv1.md (hunk 2 above)
Comment (dericed) [inline ffv1.md]: fixed in 399000b2a45d9c8a81a0f9a269048d5dca11a133

File: ffv1.md (hunk 3 above)
Comment (dericed) [inline ffv1.md]: fixed in e7c7c8abbaff660d7bf339a185ebb1f7d600a979

Comment (dericed) [inline ffv1.md]: missed handling pipe characters, updated and squashed in 904d737730804a68135ce3dc53f761093770aeb1

File: ffv1.md (hunk 4 above)
Comment (dericed) [inline ffv1.md]: Don't you mean remaining\_bits\_in\_bitstream? Updated in bf1913318a1e973e3d4107931e12d5d6c7b2ee1a.

File: ffv1.md (hunk 5 above)
Comment (dericed) [inline ffv1.md]: fixed this table and two others in 510a920c0c5ac9e54a2b1b9bb13c71d97d0b5bc4. Unfortunately I haven't found a way to remove the header row and still get a proper rendering.

Comment (dericed) [inline ffv1.md]: updated fix in 3665a3b2f93c515ba8df71265873a82eee925857

File: ffv1.md (hunk 7 above)
Comment (dericed) [inline ffv1.md]: fixed in 3665a3b2f93c515ba8df71265873a82eee925857

File: ffv1.md (hunk 6 above)
Comment (dericed) [inline ffv1.md]: fixed in 56942a6092d234c000865963d635afe7a94809dc

File: ffv1.md (hunk 9 above)
Comment (dericed) [inline ffv1.md]: fixed in c538a1d15670310001a92ef43d23dffdaa25778c

File: ffv1.md (hunk 8 above)
Comment (dericed) [inline ffv1.md]: fixed in c538a1d15670310001a92ef43d23dffdaa25778c

File: ffv1.md (hunk 10 above)
Comment (dericed) [inline ffv1.md]: fixed in f5eed1b7e42016c0be98616312cf66cb705401df

File: ffv1.md (hunk 11 above)
Comment (dericed) [inline ffv1.md]: fixed in f5eed1b7e42016c0be98616312cf66cb705401df

File: ffv1.md (hunk 9 above)
Comment (dericed) [inline ffv1.md]: updated commit via squash in f5eed1b7e42016c0be98616312cf66cb705401df

File: ffv1.md (hunk 8 above)
Comment (dericed) [inline ffv1.md]: updated via squash into f5eed1b7e42016c0be98616312cf66cb705401df

**Comment by dericed** [comment]
At this point, I think I've respond or patched according to all notes from Michael and Timothy. Please review. Let me know if moving commented code snippets from tables to code blocks is a good idea.

**Comment by dericed** [comment]
Today's added commits to the PR as in response to comments from http://ffmpeg.org/pipermail/ffmpeg-devel/2015-July/175903.html. Some remaining open issues:

- should we use syntax highlighting on code blocks?
- not yet sure how to prevent markdown tables from splitting across page breaks in the pdf output

**Pull request merged**